config/http_config.go

```diff
@@ -270,6 +277,10 @@ func NewRoundTripperFromConfig(cfg HTTPClientConfig, name string, disableKeepAli
 		if cfg.BasicAuth != nil {
 			rt = NewBasicAuthRoundTripper(cfg.BasicAuth.Username, cfg.BasicAuth.Password, cfg.BasicAuth.PasswordFile, rt)
 		}
+
+		if cfg.OAuth2 != nil {
+			rt = cfg.OAuth2.NewOAuth2RoundTripper(context.Background(), rt)
```

Comment:
we sould take the context of the request

Reply:
mmh I need to re think this

Reply:
Yeah I think the only way to do it would be taking the context as a parameter, but looking at where it's called in Prometheus (in at least one instance), the function their doesn't have the request context either. It might be good to do anyway though.